autode/calculations/executors.py

```diff
@@ -373,10 +374,14 @@ def run(self) -> None:
         method.keywords.grad = kws.GradientKeywords(self.input.keywords)
 
         self.optimiser.run(
-            species=self.molecule, method=method, n_cores=self.n_cores
+            species=self.molecule,
+            method=method,
+            n_cores=self.n_cores,
+            name=self._opt_trajectory_name,
+        )
+        self.optimiser.print_geometries(
+            self._opt_trajectory_name[:-4] + ".xyz"
```

Comment:
might be a little fragile – requires a suitable extension. I would
```python
 trajectory_name = self._opt_trajectory_name.split(".")[0] + ".xyz"
```
which should be fine if there's an extension or not

Reply:
@t-young31 I think splitting at `.` is also problematic in case there the name of the molecule has a dot in it? Because `self._opt_trajectory_name` contains `species.name`. I changed this so that it takes `[:-4]` only if it ends with ".zip" otherwise it does not do anything. What do you think of this?

Reply:
good point

> [:-4] only if it ends with ".zip" otherwise it does not do anything

👍🏼